internal/services/apimanagement/migration/api_v0_to_v1.go

```diff
@@ -0,0 +1,279 @@
+									Required: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"license": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MinItems: 1,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+					},
+					"url": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+					},
+				},
+			},
+		},
+
+		"service_url": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Computed: true,
+		},
+
+		"subscription_key_parameter_names": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"header": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+					"query": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+				},
+			},
+		},
+
+		"subscription_required": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+		},
+
+		"terms_of_service_url": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"source_api_id": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"oauth2_authorization": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"authorization_server_name": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+					"scope": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+					},
+				},
+			},
+		},
+
+		"openid_authentication": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"openid_provider_name": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+					"bearer_token_sending_methods": {
+						Type:     pluginsdk.TypeSet,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+				},
+			},
+		},
+
+		// Computed
+		"is_current": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"is_online": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"soap_pass_through": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+			Optional: true,
+		},
+
+		"version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+
+		"version_description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"version_set_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+	}
+
+	return schema
+}
+
+func (ApiV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
+	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+		apiId := fmt.Sprintf("%s;rev=%s", rawState["name"], rawState["revision"])
```

Comment:
I think these need to be cast
```suggestion
		apiId := fmt.Sprintf("%s;rev=%s", rawState["name"].(string), rawState["revision"].(string))
```